nvflare/private/fed/client/communicator.py

```diff
@@ -87,27 +87,32 @@ def __init__(
 
         self.logger = logging.getLogger(self.__class__.__name__)
 
-    def client_registration(self, client_name, servers, project_name):
+    def client_registration(self, client_name, project_name, register_data: dict, fl_ctx: FLContext):
         """Client's metadata used to authenticate and communicate.
 
         Args:
             client_name: client name
-            servers: FL servers
             project_name: FL study project name
+            register_data: customer defined client register data (in a dict)
+            fl_ctx: FLContext
 
         Returns:
             The client's token
 
         """
         local_ip = _get_client_ip()
+        shareable = Shareable()
+        shared_fl_ctx = FLContext()
+        shared_fl_ctx.set_public_props(get_serializable_data(fl_ctx).get_all_public_props())
```

Comment:
I think we should change to first get public props before checking for serializable. Otherwise it could generate unnecessary warning messages just because some private prop is not serializable.

Reply:
This is not simply reverse the order, requires more API signature changes. will improve that in a separate PR.